### DISCUSSION TITLE
How to disable http headers propagation for requests to external system?

### DISCUSSION THREAD

**Discussion by wojteksbt** [open-telemetry/opentelemetry-dotnet#5670] [Q&A] [answered]
Hi, I have a scenario where I want to use open telemetry instrumentation for http clients preserving tracing but I have some integration with system which is very strict according to headers it accepts. Therefore

1. I want to have full tracing support and headers propagation for most of the traffic
2. I want to keep trace for this external system calls
3. I don't want to send any additional headers over http to this external system

I found couple of issues on GitHub but none gave me an answer even though it doesn't look like a very unique scenario to me.

Ideally I'd like to configure it in particular HttpClient configuration defined for HttpClientFactory.

I hoped `new SocketsHttpHandler { ActivityHeadersPropagator = null })` or `DistributedContextPropagator.CreateNoOutputPropagator` is the way to go but OpenTelemetry ignores this.

I found an option with `Sdk.SetDefaultTextMapPropagator(new CompositeTextMapPropagator(Array.Empty<TextMapPropagator>()));` but even if it works (looks hacky), I need to configure that per httpClient/request and not globally.

**Comment by cijothomas** [comment, marked as answer]
See related discussion here : https://github.com/open-telemetry/opentelemetry-dotnet/issues/5667#issuecomment-2145990835
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1853

IIRC, HttpClient does not allow per request propagator, but it'd be a reasonable ask to meet this and other issues from the above.
